.devcontainer/Dockerfile

```diff
@@ -1,31 +1,52 @@
+# syntax = devthefuture/dockerfile-x
+
 ARG BASE_IMAGE=ros:humble
 
+############################################################################################################
 FROM ${BASE_IMAGE} as base
 
 USER root
 
 ENV DEBIAN_FRONTEND=noninteractive
 
+# Set the LD_PRELOAD environment variable
+ENV LD_PRELOAD=/usr/lib/libdlfaker.so:/usr/lib/libvglfaker.so
+
+# Define the workspace path
+ENV WORKSPACES=/workspaces/aoc_hri_ros2_pkg
+
 RUN apt-get update \
     && apt-get install -qq -y --no-install-recommends \
     git \
+    wget \
     python3-pip \
     python3-rosdep
 
+
+# Install all required APT packages
+INCLUDE .devcontainer/user_install.dockerfile
+
+############################################################################################################
 # get the source tree and analyse it for its package.xml only
 FROM base as sourcefilter
 COPY ./src /tmp/src
 # remove everything that isn't package.xml
 RUN find /tmp/src -type f \! -name "package.xml" -print | xargs rm -rf
 
+############################################################################################################
 # install all dependencies listed in the package.xml
 FROM base as depbuilder
 # copy the reduced source tree (only package.xml) from previous stage
 COPY --from=sourcefilter /tmp/src /tmp/src
 RUN rosdep update && apt-get update
 RUN cd /tmp/src && rosdep install --from-paths . --ignore-src -r -y && cd && rm -rf /tmp/src
 
+############################################################################################################
 FROM depbuilder as final
+
+# Copy the full src directory into the final stage
+COPY ./src ${WORKSPACES}/src
```

Comment:
I'm not sure I fully understand what the aim is here. If it's for a devcontainer for this repo, then copying in the source here seems not what is intended. This appears to be a bit of a mix between creating a fully deployable container, and installing dependencies for a dev container.

Reply:
Because we are having multi stage build, when installing the dependencies at the first stage, they do not show in the final stage, so I have to copy it. I am sure this is not the correct way to do it!!

Reply:
what do you mean, they do not show in the final stage? The decencies are resolved and part of the image. Here you are copy the source from the repositiory not the dependencies. For a devcontainer you don't copy the source as the repositiory itself is mounted in the devcontainer. So copying them only makes sense when and if one wants to make an independent docker file, to be used NOT as a devcontainer.

Reply:
what do you mean, they do not show in the final stage? The decencies are resolved and part of the image. Here you are copy the source from the repositiory not the dependencies. For a devcontainer you don't copy the source as the repositiory itself is mounted in the devcontainer. So copying them only makes sense when and if one wants to make an independent docker file, to be used NOT as a devcontainer.

Reply:
sorry I miss presented the issue. The problem that I am facing is when installing external repos in the `src` folder using this script https://github.com/ibrahimhroob/aoc_hri_ros2_pkg/blob/a2b746b5b3a1ca2ede081df32c41924266504edb/.devcontainer/user_install.dockerfile#L18-L20 they do not show after launching the devcontainer, when building it shows that it has cloned the external repos into `src` folder, however, after building the container we do not see the cloned repo in the `src` directory.

Reply:
How can I add external packages inside the src folder? The template repository is designed to handle dependency installation through a multi-stage Docker build process, which works efficiently for the existing setup. However, what is the best practice for including additional packages within the src folder that the package is dependent on and they do not have ros dep package?